File: bindings/python/pytests/test_load_ppi.py
from ensmallen_graph import EnsmallenGraph  # pylint: disable=no-name-in-module
import os

ROOT_DIR = os.path.abspath(
    os.path.dirname(__file__)
)


def test_load_ppi() -> EnsmallenGraph:
    """Test that PPI can be loaded."""
    EnsmallenGraph.from_unsorted_csv(
        edge_path=os.path.join(ROOT_DIR, "data/ppi/edges.tsv"),
        sources_column="subject",
        destinations_column="object",
        directed=False,
        weights_column="weight",
        node_path="./pytests/data/ppi/nodes.tsv",
        nodes_column="id",
        node_types_column="molecular_function",
        default_node_type="Missing"
    )

File: graph/tests/test_load_ppi.rs
extern crate graph;
use graph::*;

#[test]
fn test_load_ppi() {
    let edge_path = "tests/data/ppi/edges.tsv";
    let node_path = "tests/data/ppi/nodes.tsv";
    for directed in &[true, false] {
        let _graph = FromCsvBuilder::new(edge_path, "subject", "object", *directed, None)
            .unwrap()
            .set_weights("weight", Some(1.0))
            .load_nodes_csv(
                node_path,
                "id",
                "category",
                Some("biolink:NamedThing"),
                None,
                None,
            )
            .unwrap()
            .build()
            .unwrap();
    }
}
